### PR TITLE
[JBPM-9317] userTaskService.saveContent doesn't work

### PR DESCRIPTION
**Pull request by elguardian**
Jira:  https://issues.redhat.com/browse/JBPM-9317

### CODE REVIEW

**Comment by elguardian** [comment]
depends on 
https://github.com/kiegroup/droolsjbpm-knowledge/pull/461
https://github.com/kiegroup/jbpm/pull/1747
https://github.com/kiegroup/droolsjbpm-integration/pull/2216

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkin retest this please.

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java [hunk 1]
```diff
@@ -27,16 +29,38 @@
 
 public class SpringSecurityIdentityProvider implements IdentityProvider {
 
+    private Stack<String> contextUsers;
+    
+    public SpringSecurityIdentityProvider() {
+        contextUsers = new Stack<>();
+    }
+    @Override
+    public void setContextIdentity(String userId) {
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java:38]:
```suggestion
    public void setIdentityContext(String userId) {
```

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java [hunk 2]
```diff
@@ -27,16 +29,38 @@
...
+        contextUsers.push(userId);
+    }
+
+    @Override
+    public void removeContextIdentity() {
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java:43]:
```suggestion
    public void removeIdentityContext() {
```

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java (hunk 1 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java:38]: It is the identity of the context not the other way round.

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java (hunk 2 above)
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java:43]: It is the identity of the context not the other way round.

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java [hunk 3]
```diff
@@ -27,16 +29,38 @@
 
 public class SpringSecurityIdentityProvider implements IdentityProvider {
 
+    private Stack<String> contextUsers;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/src/main/java/org/jbpm/springboot/security/SpringSecurityIdentityProvider.java:32]: I am going to leave it like this. This does not add any value.

**Review by afalhambra — APPROVED** [review]
Great work, thanks @elguardian

**Pull request merged**